src/System.Runtime.Extensions/tests/System/IO/PathTests_Join.netcoreapp.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public class PathTests_Join : PathTestsBase
+    {
+        public static TheoryData<string, string, string> TestData_JoinTwoPaths = new TheoryData<string, string, string>
+        {
+            { "", "", "" },
+            { Sep, "", Sep },
+            { AltSep, "", AltSep },
+            { "", Sep, Sep },
+            { "", AltSep, AltSep },
+            { Sep, Sep, new string(Path.DirectorySeparatorChar, 2) },
+            { AltSep, AltSep, new string(Path.AltDirectorySeparatorChar, 2) },
+            { "a", "", "a" },
+            { "", "a", "a" },
+            { "a", "a", $"a{Sep}a" },
+            { $"a{Sep}", "a", $"a{Sep}a" },
+            { "a", $"{Sep}a", $"a{Sep}a" },
+            { $"a{Sep}", $"{Sep}a", $"a{Sep}{Sep}a" },
+            { "a", $"a{Sep}", $"a{Sep}a{Sep}" },
+            { $"a{AltSep}", "a", $"a{AltSep}a" },
+            { "a", $"{AltSep}a", $"a{AltSep}a" },
+            { $"a{Sep}", $"{AltSep}a", $"a{Sep}{AltSep}a" },
+            { $"a{AltSep}", $"{AltSep}a", $"a{AltSep}{AltSep}a" },
+            { "a", $"a{AltSep}", $"a{Sep}a{AltSep}" },
+        };
+
+        [Theory, MemberData(nameof(TestData_JoinTwoPaths))]
+        public void JoinTwoPaths(string path1, string path2, string expected)
+        {
+            Assert.Equal(expected, Path.Join(path1, path2));
+        }
+
+        [Theory, MemberData(nameof(TestData_JoinTwoPaths))]
+        public unsafe void TryJoinTwoPaths(string path1, string path2, string expected)
+        {
+            char[] output = new char[expected.Length];
+
+            Assert.True(Path.TryJoin(path1, path2, new Span<char>(output), out int written));
+            Assert.Equal(expected.Length, written);
+            if (expected.Length > 0)
+            {
+                Assert.False(Path.TryJoin(path1, path2, Span<char>.Empty, out written));
+                Assert.Equal(0, written);
+                Assert.False(Path.TryJoin(path1, path2, new Span<char>(output, 0, output.Length - 1), out written));
+                Assert.Equal(0, written);
+            }
+            Assert.Equal(expected, new string(output));
+        }
+
+        public static TheoryData<string, string, string, string> TestData_JoinThreePaths = new TheoryData<string, string, string, string>
+        {
+            { "", "", "", "" },
+            { Sep, Sep, Sep, new string(Path.DirectorySeparatorChar, 3) },
```

Comment:
nit: $"{SepSepSep}" instead of using Path.DirectorySeparatorChar to be consistent